UnityProject/Assets/Scripts/UI/Systems/AdminTools/MentorPlayerChat.cs

```diff
@@ -0,0 +1,194 @@
+				                $"-----Chat Log----- \r\n" +
+				                $" \r\n";
+				File.AppendAllText(filePath, header);
+			}
+
+			string entryName = connectedPlayer.Name;
+			if (entry.wasFromAdmin)
+			{
+				var mentorPlayer = PlayerList.Instance.GetByUserID(entry.fromUserid);
+				entryName = "[Mentor] " + mentorPlayer.Name;
+			}
+
+			DiscordWebhookMessage.Instance.AddWebHookMessageToQueue(DiscordWebhookURLs.DiscordWebhookAdminURL, entry.Message, entryName);
+
+			File.AppendAllText(filePath, $"[{DateTime.Now.ToString("O")}] {entryName}: {entry.Message}");
+		}
+
+		public void ServerGetUnreadMessages(string playerId, int currentCount, NetworkConnection requestee)
+		{
+			if (!serverMentorPlayerChatLogs.ContainsKey(playerId))
+			{
+				serverMentorPlayerChatLogs.Add(playerId, new List<AdminChatMessage>());
+			}
+
+			if (currentCount >= serverMentorPlayerChatLogs[playerId].Count)
+			{
+				return;
+			}
+
+			AdminChatUpdate update = new AdminChatUpdate();
+
+			update.messages = serverMentorPlayerChatLogs[playerId].GetRange(currentCount,
+				serverMentorPlayerChatLogs[playerId].Count - currentCount);
+
+			AdminPlayerChatUpdateMessage.SendLogUpdateToAdmin(requestee, update, playerId);
+		}
+
+		void ClientGetUnreadAdminPlayerMessages(string playerId)
+		{
+			if (!clientMentorPlayerChatLogs.ContainsKey(playerId))
+			{
+				clientMentorPlayerChatLogs.Add(playerId, new List<AdminChatMessage>());
+			}
+
+			AdminCheckMessages.Send(playerId, clientMentorPlayerChatLogs[playerId].Count);
+		}
+
+		public void ClientUpdateChatLog(string unreadMessagesJson, string playerId)
+		{
+			if (string.IsNullOrEmpty(unreadMessagesJson)) return;
+
+			if (!clientMentorPlayerChatLogs.ContainsKey(playerId))
+			{
+				clientMentorPlayerChatLogs.Add(playerId, new List<AdminChatMessage>());
+			}
+
+			var update = JsonUtility.FromJson<AdminChatUpdate>(unreadMessagesJson);
+			clientMentorPlayerChatLogs[playerId].AddRange(update.messages);
+
+			if (selectedPlayer != null && selectedPlayer.uid == playerId)
+			{
+				chatScroll.AppendChatEntries(update.messages.Cast<ChatEntryData>().ToList());
+			}
+		}
+
+		public void OnPlayerSelect(AdminPlayerEntryData playerData)
+		{
+			selectedPlayer = playerData;
+			ClientGetUnreadAdminPlayerMessages(playerData.uid);
+			if (!clientMentorPlayerChatLogs.ContainsKey(playerData.uid))
+			{
+				clientMentorPlayerChatLogs.Add(playerData.uid, new List<AdminChatMessage>());
+			}
+
+			chatScroll.LoadChatEntries(clientMentorPlayerChatLogs[playerData.uid].Cast<ChatEntryData>().ToList());
+		}
+
+		private void OnEnable()
+		{
+			chatScroll.OnInputFieldSubmit += OnInputSend;
+			if (selectedPlayer != null)
+			{
+				OnPlayerSelect(selectedPlayer);
+			}
+		}
+
+		private void OnDisable()
+		{
+			chatScroll.OnInputFieldSubmit -= OnInputSend;
+		}
+
+		public void OnInputSend(string message)
+		{
+			var mentorMsg = new AdminChatMessage
```

Comment:
Codacy found an issue: [Remove the unused local variable 'mentorMsg'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6374803)

Reply:
shush bot this is a draft pr